provider/runbook_resource_test.go

```diff
@@ -0,0 +1,325 @@
+			//{
+			//	Config: testAccRunbookResourceConfig_basic(rNameUpdated),
+			//	Check: resource.ComposeAggregateTestCheckFunc(
+			//		testAccCheckRunbookResourceExistsWithAttributes_basic("firehydrant_runbook.test_runbook"),
+			//		resource.TestCheckResourceAttrSet("firehydrant_runbook.test_runbook", "id"),
+			//		resource.TestCheckResourceAttr(
+			//			"firehydrant_runbook.test_runbook", "name", fmt.Sprintf("test-runbook-%s", rNameUpdated)),
+			//		resource.TestCheckResourceAttr(
+			//			"firehydrant_runbook.test_runbook", "type", "incident"),
+			//		resource.TestCheckResourceAttr(
+			//			"firehydrant_runbook.test_runbook", "steps.#", "1"),
+			//		resource.TestCheckResourceAttr(
+			//			"firehydrant_runbook.test_runbook", "steps.0.name", "Create Incident Channel"),
+			//		resource.TestCheckResourceAttrSet(
+			//			"firehydrant_runbook.test_runbook", "steps.0.action_id"),
+			//	),
+			//},
+		},
+	})
+}
+
+func TestAccRunbookResourceImport_basic(t *testing.T) {
+	rName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRunbookResourceConfig_basic(rName),
+			},
+			{
+				ResourceName:      "firehydrant_runbook.test_runbook",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccRunbookResourceImport_allAttributes(t *testing.T) {
+	rName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRunbookResourceConfig_update(rName),
+			},
+			{
+				ResourceName:      "firehydrant_runbook.test_runbook",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckRunbookResourceExistsWithAttributes_basic(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		runbookResource, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+		if runbookResource.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		client, err := firehydrant.NewRestClient(os.Getenv("FIREHYDRANT_API_KEY"))
+		if err != nil {
+			return err
+		}
+
+		runbookResponse, err := client.Runbooks().Get(context.TODO(), runbookResource.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		expected, got := runbookResource.Primary.Attributes["name"], runbookResponse.Name
+		if expected != got {
+			return fmt.Errorf("Unexpected name. Expected: %s, got: %s", expected, got)
+		}
+
+		expected, got = runbookResource.Primary.Attributes["type"], fmt.Sprintf("%s", runbookResponse.Type)
+		if expected != got {
```

Comment:
Might be that I am just so use to Jest and Rspec but interesting to see that you pretty much have to manage your own errors within your test in Go. I would think that something like is common enough to be a function.
```go
fmt.Errorf("Unexpected type. Expected: %s, got: %s", expected, got)
```